app/models/embedded_ansible_worker.rb

```diff
@@ -17,13 +22,35 @@ def start_runner
         Thread.exit
       end
     end
-    nil # return no pid
+
+    t[:worker_class] = self.class.name
+    t[:worker_id]    = id
```

Comment:
Here we're setting thread instance variables in the Thread object so we can locate the thread for this worker later.